Firestore/core/src/firebase/firestore/util/executor_libdispatch.mm

```diff
@@ -60,16 +60,19 @@ void DispatchAsync(const dispatch_queue_t queue, std::function<void()>&& work) {
 }
 
 void DispatchSync(const dispatch_queue_t queue, std::function<void()> work) {
-  HARD_ASSERT(
-      GetCurrentQueueLabel() != GetQueueLabel(queue),
-      "Calling DispatchSync on the current queue will lead to a deadlock.");
+  if (GetCurrentQueueLabel() == GetQueueLabel(queue)) {
```

Comment:
Note: this function is very similar to `RunSynchronized` below; it looks like `RunSynchronized` can now be removed (or perhaps this change can be avoided by using `RunSynchronized` -- though having a non-deadlocking `DispatchSync` is probably preferable).